pos-frontend/src/components/SignUp.jsx

```diff
@@ -0,0 +1,223 @@
+import { Container, Grid, CssBaseline, Typography, TextField, Box, Avatar, Button, Snackbar, Alert } from '@mui/material'
+import { Link, useNavigate } from "react-router-dom"
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined'
+import { useForm } from "react-hook-form"
+import axios from "axios"
+import { useState } from 'react'
+
+const initialState = {
+  open: false,
+  status: '',
+  message: ''
+}
+
+const SignUp = (props) => {
+
+  const { register , handleSubmit, formState: { errors }, watch, getValues, reset, setFocus, setError } = useForm();
+  const [ message, setMessage ] = useState(initialState);
+  const navigate = useNavigate()
+
+  const onSubmit = () => {
+    const userData = getValues();
+
+    axios.post("http://localhost:5000/api/auth/signup", userData).then((res) => {
```

Comment:
why is the api url here ... move it to a common file

Reply:
Moved to common file and using is as a const